tap_salesforce/__init__.py

```diff
@@ -159,15 +161,16 @@ def do_discover(sf):
 
         found_id_field = False
 
+        expected_pk_field = PK_OVERRIDES.get(sobject_name, "Id")
+
         # Loop over the object's fields
         for f in fields:
             field_name = f['name']
 
-            if field_name == "Id":
+            if field_name == expected_pk_field:
                 found_id_field = True
```

Comment:
I guess this doesn't really matter but I think we should rename this to be consistent with the new variable